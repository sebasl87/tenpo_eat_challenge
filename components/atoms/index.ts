export { default as RestaurantsLayout } from './RestaurantsLayout';
