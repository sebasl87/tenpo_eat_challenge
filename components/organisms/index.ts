export { default as Header } from "./Header";
export { default as HeaderLocation } from "./HeaderLocation";
