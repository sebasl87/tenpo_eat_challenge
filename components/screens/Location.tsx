import React, { useEffect, useState } from 'react'
import { Platform, StyleSheet, View } from 'react-native';
import MapView, { Marker, PROVIDER_GOOGLE } from 'react-native-maps';
import { check, request, PERMISSIONS, RESULTS } from "react-native-permissions" // ðŸ‘ˆ
import Geolocation from "react-native-geolocation-service"

export interface IGoogleMaps {
    latitude: number;
    longitude: number;
}

function Location() {
    const [location, setLocation] = useState<IGoogleMaps>()

    const handleLocationPermission = async () => {
        let permissionCheck = '';
        if (Platform.OS === 'ios') {
            permissionCheck = await check(PERMISSIONS.IOS.LOCATION_WHEN_IN_USE);

            if (
                permissionCheck === RESULTS.BLOCKED ||
                permissionCheck === RESULTS.DENIED
            ) {
                const permissionRequest = await request(
                    PERMISSIONS.IOS.LOCATION_WHEN_IN_USE,
                );
                permissionRequest === RESULTS.GRANTED
                    ? console.warn('Location permission granted.')
                    : console.warn('location permission denied.');
            }
        }

        if (Platform.OS === 'android') {
            permissionCheck = await check(PERMISSIONS.ANDROID.ACCESS_FINE_LOCATION);

            if (
                permissionCheck === RESULTS.BLOCKED ||
                permissionCheck === RESULTS.DENIED
            ) {
                const permissionRequest = await request(
                    PERMISSIONS.ANDROID.ACCESS_FINE_LOCATION,
                );
                permissionRequest === RESULTS.GRANTED
                    ? console.warn('Location permission granted.')
                    : console.warn('location permission denied.');
            }
        }
    };

    useEffect(() => {
        handleLocationPermission()
    }, [])

    useEffect(() => {
        Geolocation.getCurrentPosition(
            position => {
                const { latitude, longitude } = position.coords
                setLocation({ latitude, longitude })
            },
            error => {
                console.log(error.code, error.message)
            },
            { enableHighAccuracy: true, timeout: 15000, maximumAge: 10000 }
        )
    }, [])

    return (
        <View
            style={styles.mapStyle}
        >
            {location && (<MapView
                style={styles.mapStyle}
                provider={PROVIDER_GOOGLE}
                initialRegion={{
                    latitude: location.latitude,
                    longitude: location.longitude,
                    latitudeDelta: 0.0922,
                    longitudeDelta: 0.0421,
                }}
                showsUserLocation={true}
            >
                <Marker
                    image={require("../../assets/icons/pointer.png")}
                    style={{ height: 50 }}
                    coordinate={{
                        latitude: location.latitude,
                        longitude: location.longitude,
                    }}
                    title={'Test Marker'}
                />
            </MapView>)}
        </View>
    );
}
const styles = StyleSheet.create({

    container: {
        position: 'absolute',
        top: 0,
        left: 0,
        right: 0,
        bottom: 0,
        alignItems: 'center',
        justifyContent: 'flex-end',
    },
    mapStyle: {
        position: 'absolute',
        top: 0,
        left: 0,
        right: 0,
        bottom: 0,
    },
});
export default Location