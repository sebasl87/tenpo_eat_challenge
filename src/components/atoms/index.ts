import AddressLayout from './AddressLayout';
export { default as RestaurantsLayout } from './RestaurantsLayout';
export { default as CategoriesLayout } from './CategoriesLayout';
export { default as HeaderFavorites } from './HeaderFavorites';
export { default as AddressLayout } from './AddressLayout';
