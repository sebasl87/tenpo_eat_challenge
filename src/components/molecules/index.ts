export { default as DetailCard } from './DetailCard';
export { default as DiscountCircle } from './DiscountCircle';
export { default as LogoRounded } from './LogoRounded';
export { default as DetailFavorites } from './DetailFavorites';
