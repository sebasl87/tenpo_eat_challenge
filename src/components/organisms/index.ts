export { default as Header } from "./Header";
export { default as HeaderLocation } from "./HeaderLocation";
export { default as ContainerMapView } from "./ContainerMapView";
export { default as RestaurantRow } from "./RestaurantRow";
export { default as ContainerAddressData } from "./ContainerAddressData";
export { default as CategoriesRow } from "./CategoriesRow";
export { default as FavoritesRow } from "./FavoritesRow";
export { default as HomeLocation } from './HomeLocation';
