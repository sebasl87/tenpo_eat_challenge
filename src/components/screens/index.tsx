export { default as HomeScreen } from "./HomeScreen";
export { default as Location } from "./Location";
export { default as RestaurantDetail } from "./RestaurantDetail";