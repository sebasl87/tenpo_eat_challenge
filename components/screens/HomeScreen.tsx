import React from 'react'
import { Restaurant } from '../organisms'


function HomeScreen() {
    return (
        <>
            <Restaurant />
        </>
    );
}

export default HomeScreen;