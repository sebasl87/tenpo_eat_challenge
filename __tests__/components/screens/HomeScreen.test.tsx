import React from "react"
import renderer from "react-test-renderer";

import { HomeScreen } from "../../../src/components/screens"

const mockDispatch = jest.fn();

jest.mock('react-redux', () => ({
    ...jest.requireActual('react-redux'),
    useSelector: jest.fn(() => ({
        restaurants: {
            restoList: [
                {
                    id: 1,
                    uri: 'https://i.imgur.com/1PLcyBL.png',
                    uri_large: 'https://i.imgur.com/25zvKsX.png',
                    discount: '50',
                    restaurantName: 'Mcdonalds',
                    rate: '3.5',
                    lapseTime: '10-50',
                    bgColor: '',
                },
            ],
        },
    })),
    useDispatch: () => mockDispatch,
}));
describe("<HomeScreen />", () => {

    it("should render successfully with Restaurant_mock", () => {
        const restaurantSection = renderer
            .create(<HomeScreen />)
            .toJSON();

        expect(restaurantSection).toMatchSnapshot();
    });
})