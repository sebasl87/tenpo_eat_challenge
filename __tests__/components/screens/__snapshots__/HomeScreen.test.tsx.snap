// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeScreen /> should render successfully with Restaurant_mock 1`] = `
[
  <View
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "#D4F9F5",
          "borderTopLeftRadius": 28,
          "borderTopRightRadius": 28,
          "paddingBottom": 18,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 20,
        },
      ]
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <Image
          source={
            {
              "testUri": "../../../assets/icons/location.png",
            }
          }
          style={
            {
              "height": 18,
              "marginRight": 6,
              "paddingTop": 30,
              "width": 24,
            }
          }
        />
        <Text
          numberOfLines={1}
          style={
            [
              {
                "color": "#008f7e",
                "fontSize": 16,
                "marginLeft": 8,
                "maxWidth": 300,
              },
              {
                "fontFamily": "Gotham-Light",
              },
            ]
          }
        >
          Agregar direcci√≥n de la entrega
        </Text>
      </View>
    </View>
  </View>,
  <View
    style={
      [
        {
          "backgroundColor": "#fff",
          "borderColor": "#00000029",
          "borderStyle": "solid",
          "borderTopLeftRadius": 28,
          "borderTopRightRadius": 28,
          "borderWidth": 1,
          "height": "100%",
          "paddingBottom": 0,
          "paddingLeft": 16,
          "paddingRight": 0,
          "paddingTop": 42,
        },
      ]
    }
  >
    <RCTScrollView>
      <View>
        <View>
          <Text
            style={
              [
                {
                  "color": "#333333",
                  "fontSize": 18,
                  "marginBottom": 6,
                },
                {
                  "fontFamily": "Gotham-Bold",
                },
              ]
            }
          >
            RESTAURANTES
          </Text>
          <RCTScrollView
            horizontal={true}
          >
            <View>
              <View
                style={
                  {
                    "flexDirection": "row",
                  }
                }
              />
            </View>
          </RCTScrollView>
        </View>
        <View
          style={
            {
              "marginTop": 20,
            }
          }
        >
          <Text
            style={
              [
                {
                  "color": "#333333",
                  "fontSize": 18,
                  "marginBottom": 6,
                  "marginTop": 32,
                },
                {
                  "fontFamily": "Gotham-Bold",
                },
              ]
            }
          >
            CATEGORIAS
          </Text>
          <RCTScrollView
            horizontal={true}
          >
            <View>
              <View
                style={
                  {
                    "flexDirection": "row",
                  }
                }
              >
                <View
                  style={
                    {
                      "marginRight": 6,
                    }
                  }
                >
                  <View
                    style={
                      [
                        {
                          "alignContent": "center",
                          "alignItems": "center",
                          "display": "flex",
                          "height": 90,
                          "width": 168,
                        },
                      ]
                    }
                  >
                    <Text
                      style={
                        [
                          {
                            "color": "#fff",
                            "fontSize": 16,
                            "letterSpacing": 1,
                            "marginTop": 30,
                            "position": "absolute",
                            "zIndex": 2,
                          },
                          {
                            "fontFamily": "Gotham-Bold",
                          },
                        ]
                      }
                    >
                      HAMBURGUESAS
                    </Text>
                    <View
                      accessibilityIgnoresInvertColors={true}
                      style={
                        [
                          {
                            "height": 80,
                            "width": 174,
                          },
                        ]
                      }
                    >
                      <Image
                        source={
                          {
                            "testUri": "../../../assets/imgs/hamburguer.png",
                          }
                        }
                        style={
                          [
                            {
                              "bottom": 0,
                              "left": 0,
                              "position": "absolute",
                              "right": 0,
                              "top": 0,
                            },
                            {
                              "height": 80,
                              "width": 174,
                            },
                            undefined,
                          ]
                        }
                      />
                    </View>
                  </View>
                </View>
                <View
                  style={
                    {
                      "marginRight": 6,
                    }
                  }
                >
                  <View
                    style={
                      [
                        {
                          "alignContent": "center",
                          "alignItems": "center",
                          "display": "flex",
                          "height": 90,
                          "width": 168,
                        },
                      ]
                    }
                  >
                    <Text
                      style={
                        [
                          {
                            "color": "#fff",
                            "fontSize": 16,
                            "letterSpacing": 1,
                            "marginTop": 30,
                            "position": "absolute",
                            "zIndex": 2,
                          },
                          {
                            "fontFamily": "Gotham-Bold",
                          },
                        ]
                      }
                    >
                      ITALIANA
                    </Text>
                    <View
                      accessibilityIgnoresInvertColors={true}
                      style={
                        [
                          {
                            "height": 80,
                            "width": 174,
                          },
                        ]
                      }
                    >
                      <Image
                        source={
                          {
                            "testUri": "../../../assets/imgs/italian.png",
                          }
                        }
                        style={
                          [
                            {
                              "bottom": 0,
                              "left": 0,
                              "position": "absolute",
                              "right": 0,
                              "top": 0,
                            },
                            {
                              "height": 80,
                              "width": 174,
                            },
                            undefined,
                          ]
                        }
                      />
                    </View>
                  </View>
                </View>
                <View
                  style={
                    {
                      "marginRight": 6,
                    }
                  }
                >
                  <View
                    style={
                      [
                        {
                          "alignContent": "center",
                          "alignItems": "center",
                          "display": "flex",
                          "height": 90,
                          "width": 168,
                        },
                      ]
                    }
                  >
                    <Text
                      style={
                        [
                          {
                            "color": "#fff",
                            "fontSize": 16,
                            "letterSpacing": 1,
                            "marginTop": 30,
                            "position": "absolute",
                            "zIndex": 2,
                          },
                          {
                            "fontFamily": "Gotham-Bold",
                          },
                        ]
                      }
                    >
                      PIZZAS
                    </Text>
                    <View
                      accessibilityIgnoresInvertColors={true}
                      style={
                        [
                          {
                            "height": 80,
                            "width": 174,
                          },
                        ]
                      }
                    >
                      <Image
                        source={
                          {
                            "testUri": "../../../assets/imgs/pizza.png",
                          }
                        }
                        style={
                          [
                            {
                              "bottom": 0,
                              "left": 0,
                              "position": "absolute",
                              "right": 0,
                              "top": 0,
                            },
                            {
                              "height": 80,
                              "width": 174,
                            },
                            undefined,
                          ]
                        }
                      />
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </RCTScrollView>
        </View>
        <View
          style={
            {
              "marginBottom": 80,
              "marginTop": 40,
            }
          }
        >
          <Text
            style={
              [
                {
                  "color": "#333333",
                  "fontSize": 18,
                  "marginBottom": 6,
                },
                {
                  "fontFamily": "Gotham-Bold",
                },
              ]
            }
          >
            TUS FAVORITOS
          </Text>
          <RCTScrollView
            horizontal={true}
          >
            <View>
              <View
                style={
                  {
                    "flexDirection": "row",
                    "marginBottom": 80,
                  }
                }
              >
                <View
                  style={
                    {
                      "marginRight": 12,
                    }
                  }
                >
                  <View
                    style={
                      [
                        {
                          "display": "flex",
                          "height": 102,
                          "width": 260,
                        },
                      ]
                    }
                  >
                    <Image
                      source={
                        {
                          "testUri": "../../../assets/imgs/mc.png",
                        }
                      }
                      style={
                        [
                          {
                            "height": 44,
                            "marginLeft": 11,
                            "marginTop": 9,
                            "position": "absolute",
                            "width": 44,
                            "zIndex": 1,
                          },
                        ]
                      }
                    />
                    <View
                      accessibilityIgnoresInvertColors={true}
                      style={
                        [
                          {
                            "height": 102,
                            "width": 260,
                          },
                        ]
                      }
                    >
                      <Image
                        source={
                          {
                            "testUri": "../../../assets/imgs/mcBack.png",
                          }
                        }
                        style={
                          [
                            {
                              "bottom": 0,
                              "left": 0,
                              "position": "absolute",
                              "right": 0,
                              "top": 0,
                            },
                            {
                              "height": 102,
                              "width": 260,
                            },
                            undefined,
                          ]
                        }
                      />
                    </View>
                  </View>
                  <View
                    style={
                      [
                        {
                          "borderBottomLeftRadius": 16,
                          "borderBottomRightRadius": 16,
                          "borderColor": "#EDEDED",
                          "borderStyle": "solid",
                          "borderWidth": 1,
                          "display": "flex",
                          "maxWidth": 260,
                          "paddingBottom": 8,
                          "paddingLeft": 8,
                          "paddingRight": 8,
                          "paddingTop": 8,
                          "width": "100%",
                        },
                      ]
                    }
                  >
                    <View
                      style={
                        [
                          {
                            "alignItems": "center",
                            "display": "flex",
                            "flexDirection": "row",
                            "justifyContent": "space-between",
                            "marginBottom": 3,
                            "maxWidth": 260,
                            "width": "100%",
                          },
                        ]
                      }
                    >
                      <Text
                        style={
                          [
                            {
                              "color": "#333333",
                              "fontSize": 13,
                            },
                            {
                              "fontFamily": "Gotham-Medium",
                            },
                          ]
                        }
                      >
                        Combo hamburguesa Bigmac
                      </Text>
                      <View
                        style={
                          [
                            {
                              "alignItems": "center",
                              "display": "flex",
                              "flexDirection": "row",
                            },
                          ]
                        }
                      >
                        <Image
                          source={
                            {
                              "testUri": "../../../assets/imgs/starmini.png",
                            }
                          }
                          style={
                            [
                              {
                                "height": 16,
                                "marginRight": 2,
                                "width": 16,
                              },
                            ]
                          }
                        />
                        <Text
                          style={
                            [
                              {
                                "color": "#333333",
                                "fontSize": 12,
                              },
                              {
                                "fontFamily": "Gotham-Medium",
                              },
                            ]
                          }
                        >
                          3.5
                        </Text>
                      </View>
                    </View>
                    <View
                      style={
                        [
                          {
                            "alignItems": "center",
                            "display": "flex",
                            "flexDirection": "row",
                            "justifyContent": "space-between",
                            "maxWidth": 260,
                            "width": "100%",
                          },
                        ]
                      }
                    >
                      <Text
                        style={
                          [
                            {
                              "color": "#00baa4",
                              "fontSize": 13,
                            },
                            {
                              "fontFamily": "Gotham-Bold",
                            },
                          ]
                        }
                      >
                        Mcdonalds
                      </Text>
                      <View>
                        <Text
                          style={
                            [
                              {
                                "color": "#333333",
                                "fontSize": 12,
                              },
                              {
                                "fontFamily": "Gotham-Medium",
                              },
                            ]
                          }
                        >
                          10-50
                           min
                        </Text>
                      </View>
                    </View>
                  </View>
                </View>
                <View
                  style={
                    {
                      "marginRight": 12,
                    }
                  }
                >
                  <View
                    style={
                      [
                        {
                          "display": "flex",
                          "height": 102,
                          "width": 260,
                        },
                      ]
                    }
                  >
                    <Image
                      source={
                        {
                          "testUri": "../../../assets/imgs/melt.png",
                        }
                      }
                      style={
                        [
                          {
                            "height": 44,
                            "marginLeft": 11,
                            "marginTop": 9,
                            "position": "absolute",
                            "width": 44,
                            "zIndex": 1,
                          },
                        ]
                      }
                    />
                    <View
                      accessibilityIgnoresInvertColors={true}
                      style={
                        [
                          {
                            "height": 102,
                            "width": 260,
                          },
                        ]
                      }
                    >
                      <Image
                        source={
                          {
                            "testUri": "../../../assets/imgs/meltBack.png",
                          }
                        }
                        style={
                          [
                            {
                              "bottom": 0,
                              "left": 0,
                              "position": "absolute",
                              "right": 0,
                              "top": 0,
                            },
                            {
                              "height": 102,
                              "width": 260,
                            },
                            undefined,
                          ]
                        }
                      />
                    </View>
                  </View>
                  <View
                    style={
                      [
                        {
                          "borderBottomLeftRadius": 16,
                          "borderBottomRightRadius": 16,
                          "borderColor": "#EDEDED",
                          "borderStyle": "solid",
                          "borderWidth": 1,
                          "display": "flex",
                          "maxWidth": 260,
                          "paddingBottom": 8,
                          "paddingLeft": 8,
                          "paddingRight": 8,
                          "paddingTop": 8,
                          "width": "100%",
                        },
                      ]
                    }
                  >
                    <View
                      style={
                        [
                          {
                            "alignItems": "center",
                            "display": "flex",
                            "flexDirection": "row",
                            "justifyContent": "space-between",
                            "marginBottom": 3,
                            "maxWidth": 260,
                            "width": "100%",
                          },
                        ]
                      }
                    >
                      <Text
                        style={
                          [
                            {
                              "color": "#333333",
                              "fontSize": 13,
                            },
                            {
                              "fontFamily": "Gotham-Medium",
                            },
                          ]
                        }
                      >
                        Pizza Mediana 3 ingredientes
                      </Text>
                      <View
                        style={
                          [
                            {
                              "alignItems": "center",
                              "display": "flex",
                              "flexDirection": "row",
                            },
                          ]
                        }
                      >
                        <Image
                          source={
                            {
                              "testUri": "../../../assets/imgs/starmini.png",
                            }
                          }
                          style={
                            [
                              {
                                "height": 16,
                                "marginRight": 2,
                                "width": 16,
                              },
                            ]
                          }
                        />
                        <Text
                          style={
                            [
                              {
                                "color": "#333333",
                                "fontSize": 12,
                              },
                              {
                                "fontFamily": "Gotham-Medium",
                              },
                            ]
                          }
                        >
                          4.5
                        </Text>
                      </View>
                    </View>
                    <View
                      style={
                        [
                          {
                            "alignItems": "center",
                            "display": "flex",
                            "flexDirection": "row",
                            "justifyContent": "space-between",
                            "maxWidth": 260,
                            "width": "100%",
                          },
                        ]
                      }
                    >
                      <Text
                        style={
                          [
                            {
                              "color": "#00baa4",
                              "fontSize": 13,
                            },
                            {
                              "fontFamily": "Gotham-Bold",
                            },
                          ]
                        }
                      >
                        MELT pizzas
                      </Text>
                      <View>
                        <Text
                          style={
                            [
                              {
                                "color": "#333333",
                                "fontSize": 12,
                              },
                              {
                                "fontFamily": "Gotham-Medium",
                              },
                            ]
                          }
                        >
                          10-60
                           min
                        </Text>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </RCTScrollView>
        </View>
      </View>
    </RCTScrollView>
  </View>,
]
`;
